server/src/test/java/org/elasticsearch/cli/MultiCommandTests.java

```diff
@@ -102,4 +126,27 @@ public void testSubcommandArguments() throws Exception {
         assertFalse(output, output.contains("command1"));
         assertTrue(output, output.contains("Arguments: [foo, bar]"));
     }
+
+    public void testClose() throws Exception {
+        DummySubCommand subCommand1 = new DummySubCommand();
+        DummySubCommand subCommand2 = new DummySubCommand();
+        multiCommand.subcommands.put("command1", subCommand1);
+        multiCommand.subcommands.put("command2", subCommand2);
+        multiCommand.close();
+        assertTrue("MultiCommand must have been closed when close method is invoked", multiCommand.closed.get());
+        assertTrue("SubCommand1 must have been closed when close method is invoked", subCommand1.closed.get());
+        assertTrue("SubCommand2 must have been closed when close method is invoked", subCommand2.closed.get());
+    }
+
+    public void testCloseWhenSubCommandCloseThrowsException() {
+        boolean throwExceptionWhenClosed = true;
```

Comment:
It would be nice to test the other scenarios or add some randomization to this test in terms of which subcommand(s) throws an exception:

```java
final boolean command1Throws = randomBoolean();
final boolean command2Throws = randomBoolean();
...
assertTrue(command1Throws ^ subCommand1.closed.get());
assertTrue(command2Throws ^ subCommand2.closed.get());
```